web/modules/custom/joinup_invite/joinup_invite.behat.inc

```diff
@@ -74,11 +142,103 @@ class JoinupInviteSubContext extends DrupalSubContextBase implements DrupalSubCo
         'invitation_bundle' => 'discussion',
         'message_template' => InviteToDiscussionForm::TEMPLATE_DISCUSSION_INVITE,
       ],
+      'collection' => [
+        'invitation_bundle' => 'group',
+        'message_template' => InviteToGroupForm::TEMPLATE_GROUP_INVITE,
+      ],
     ];
 
     return $mapping[$entity_type];
   }
 
+  /**
+   * Loads an unchanged invitation.
+   *
+   * @param \Drupal\Core\Entity\ContentEntityInterface $entity
+   *   The entity related to the invitation.
+   * @param \Drupal\Core\Session\AccountInterface $recipient
+   *   The recipient related to the invitation.
+   * @param string $bundle
+   *   The bundle of the invitation.
+   *
+   * @return \Drupal\joinup_invite\Entity\InvitationInterface
+   *   The invitation object.
+   *
+   * @throws \Exception
+   *   Thrown when an invitation is not found.
+   */
+  protected function loadByEntityAndUserUnchanged(ContentEntityInterface $entity, AccountInterface $recipient, string $bundle = ''): ?InvitationInterface {
+    $storage = \Drupal::entityTypeManager()->getStorage('invitation');
+    $query = $storage->getQuery();
+    $query->condition('entity_type', $entity->getEntityTypeId())
+      ->condition('entity_id', $entity->id())
+      ->condition('recipient_id', $recipient->id())
+      ->condition('bundle', $bundle);
+    $results = $query->execute();
+    if (empty($results)) {
+      throw new Exception("No invitation was found.");
+    }
+
+    $iid = reset($results);
+    /** @var \Drupal\joinup_invite\Entity\InvitationInterface $invitation */
+    $invitation = $storage->loadUnchanged($iid);
```

Comment:
I only kept it like this for the IDE. Since the invitation storage is simply the entity storage, the return complains that an EntityInterface is returned when an Invitation entity is expected.
While we could simply set it to return an Entity interface, this would result in other methods not being recognized.
By having the `/** @var \type $variable */` on top, fixes that issue in the IDE.